### PR TITLE
ABP CLI Replace Packages with Previews

### PR DESCRIPTION
**Pull request by yekalkan**
resolves https://github.com/abpframework/abp/issues/2837

### CODE REVIEW

File: framework/src/Volo.Abp.Cli.Core/Volo/Abp/Cli/ProjectModification/PackageSourceSwitcher.cs [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+
+            var versionAttr = doc.CreateAttribute("Version");
+            versionAttr.Value = newVersion;
+            newNode.Attributes.Append(versionAttr);
+            return newNode;
+        }
+
+        private string GetFormattedDate()
+        {
+            var formattedDate = "";
```
Comment (soyhanbeyazit) [inline framework/src/Volo.Abp.Cli.Core/Volo/Abp/Cli/ProjectModification/PackageSourceSwitcher.cs]: Is there any specific reason to have a method like this? AFAIK, `DateTime.Now.ToString("yyyyMMdd")` should be enough to get an output like this `20200221`

**Comment by hikalkan** [comment]
Does this cover NPM packages?

**Comment by yekalkan** [comment]
@hikalkan No

**Comment by hikalkan** [comment]
>@hikalkan No

then it should do.

**Comment by maliming** [comment]
hi @yekalkan Has this PR been completed?

**Comment by yekalkan** [comment]
> hi @yekalkan Has this PR been completed?

No. I'm waiting for npm-preview packages.

**Pull request merged**